site/docs/concepts/optimization-techniques/kvcache-eviction-algorithm.md

```diff
@@ -3,3 +3,55 @@ sidebar_position: 2
 ---
 
 # KVCache Token Eviction Algorithm
+
+
+## Overview
+The cache eviction algorithm is designed to manage KV (Key-Value) cache memory for large language models (LLMs) during text generation. It determines which blocks of tokens should be evicted from the KV cache based on importance scores calculated from attention scores across different attention layers.
+
+## Conceptual Model
+The KV cache for each sequence is divided into three logical areas:
+
+![KV cache layout with cache eviction](./images/kv-cache-areas-diagram.svg)
+
+* Start Area: Initial tokens that are never evicted
+* Evictable Area: Tokens that can be evicted based on importance scores
+* Recent Area: Most recent tokens that are preserved (never evicted)
+
+The sizes of all three areas can be configured by modifying corresponding fields in a `CacheEvictionConfig` struct, which itself is a part of the pipeline-wide `SchedulerConfig`.
+As the generation starts, the blocks in respective logical areas are filled token-by-token, and once at least one block past the "recent" area is filled, eviction may take place. 
+The tokens are evicted based on accumulated importance scores following the [H2O](https://arxiv.org/abs/2306.14048) approach.
+The scores are accumulated throughout the entire generation process and their weighting may be changed by adjusting the `CacheEvictionConfig.aggregation_mode` parameter.
+Eviction occurs with a block-wise granularity, and only the completely filled blocks from the "evictable" area are evicted.
+
+This approach allows LLMs to handle long sequences efficiently by keeping the most contextually important tokens in the cache while evicting those of lesser importance.
+The downside of the eviction procedure is potential loss of generation accuracy, since the cache no longer contains the entire context for the generation, but only the most "important" token blocks.
+The user can adjust the individual sizes of the eviction sub-areas to hit the optimal point of accuracy/memory usage tradeoff in their particular case.
+
+Note that currently the eviction only starts after the full prompt has been processed, i.e. no eviction takes place during the prefill phase.
+This means that for longer prompt sizes the maximum cache usage may exceed the limit defined by the `CacheEvictionConfig` parameters. 
+
+After the prefill phase, however, the maximum cache occupancy for each sequence currently being processed is strictly limited by the combined sizes of the 3 areas described above. 
+`CacheEvictionConfig.get_max_cache_size_after_eviction()` can be queried to get this cache size limit in tokens.
+
+
+## Sample - impact of cache eviction on possible generation length and prompt throughput
+[limit_checker.py](https://github.com/openvinotoolkit/openvino.genai/tree/master/tools/continuous_batching/limit_checker.py) can be used to visualize the impact of the cache eviction algorithm on the end performance of the generation pipeline.
```

Comment:
I assume it works on Linux only. Let's add a note about it or specify it at the beginning of the script or in the description/help.

Reply:
There is no such limitation in the script code, and I think that the pipeline itself also works on Windows now.